application/common/models/User.php

```diff
@@ -914,4 +914,19 @@ public function afterFind()
 
         parent::afterFind();
     }
+
+    public function assessPassword(string $newPassword): bool
+    {
+        $password = new Password();
+
+        $password->user_id = $this->id;
+        $password->password = $newPassword;
+
+        if (! $password->validate()) {
+            $this->addErrors($password->getErrors());
```

Comment:
Are the errors returned by `getErrors()` associated with specific attribute names? I didn't know if taking a `Password` record's errors and setting them on a `User` record would have trouble due to there being different attribute names.

Reply:
Seems to work. Also, I just took a glance at the Yii code to confirm it doesn't check the value of the array key.